### PR TITLE
Lambda for crud access

### PR DESCRIPTION
**Pull request by alex-melio**
In this PR:

* Makefile improvement supporting two targets (can probably be improved further to avoid redundant work), and copying of ormconfig file and node_modules directory into artifact directory (unsure how it worked before that)
* New lambda doing crud access
* Script for database population (only for dev)
* node run script for db population was added, script for cleaning was modified to avoid removal of node modules since it was called after each build cycle

### CODE REVIEW

File: packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    isBase64Encoded: false,
+    statusCode,
+  };
+};
+
+const formatError = function (body: string, statusCode = 500) {
+  return formatResponse(body, statusCode);
+};
+
+const upsert = async function (vendorObject: Vendor) {
```
Comment (alex-melio) [inline packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts:25]: simply `vendor` will also do

File: packages/vendor-directory-api/Makefile [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 	npm run compile
 	npm prune --production
-	cp -a out package.json package-lock.json $(ARTIFACTS_DIR)
+	cp -a out package.json package-lock.json node_modules ormconfig.json $(ARTIFACTS_DIR)
 
 build-MatchRawVendorFunction: default
-	
-.PHONY: build-MatchRawVendorFunction default
\ No newline at end of file
+build-DirectoryVendorFunction: default
```
Comment (alex-melio) [inline packages/vendor-directory-api/Makefile:14]: break the process into two parts, compile (will happen once), and copy (will happen as many as we have functions)

File: packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      'Content-Type': 'application/json',
+    },
+    isBase64Encoded: false,
+    statusCode,
+  };
+};
+
+const formatError = function (body: string, statusCode = 500) {
+  return formatResponse(body, statusCode);
+};
```
Comment (alex-melio) [inline packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts:23]: move this to utils

File: packages/vendor-directory-api/template.yml [hunk 4]
```diff
@@ -100,7 +100,7 @@ Resources:
     Type: AWS::RDS::DBCluster
```
Comment (alex-melio) [inline packages/vendor-directory-api/template.yml:100]: should we remove this clause?
cluster is no longer managed here

File: packages/vendor-directory-api/openapi.yaml [hunk 5]
```diff
@@ -309,6 +263,30 @@ components:
     Address:
       type: string
       description: Vendor's identifying physical check destination address
+    PutDirectoryVendor:
```
Comment (alex-melio) [inline packages/vendor-directory-api/openapi.yaml:266]: consolidate some of the vendor types

File: packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  return { hits: result.length, vendors: result };
+};
+
+const deleteVendorByID = async function (vdid: string) {
+  const result = await Vendor.delete(vdid);
+  return { deleted: result.affected };
+};
+
+export const handler: APIGatewayProxyHandler = (async (event) => {
+  if (event.multiValueHeaders === undefined) {
```
Comment (shaikegross) [inline packages/vendor-directory-api/src/handlers/directory-vendor-handler/index.ts:41]: Can you move this to a function called verifyPayloadVersion.

Code there could be more descriptive and could reduce the need in extensive comments.

For example: `isVersion1 = !!event.multiValueHeaders`

File: packages/vendor-directory-api/Makefile (hunk 2 above)
Comment (shaikegross) [reply packages/vendor-directory-api/Makefile:14]: Agreed not critical now

File: packages/vendor-directory-api/template.yml (hunk 4 above)
Comment (shaikegross) [reply packages/vendor-directory-api/template.yml:100]: For now no as aws_iam isn't fully baked

File: packages/vendor-directory-api/template.yml [hunk 7]
```diff
@@ -151,3 +151,20 @@ Resources:
...
+
+  DirectoryVendorFunction:
+    Type: AWS::Serverless::Function
+    Metadata:
+      BuildMethod: makefile
+    Properties:
+      CodeUri: ./
+      Handler:  out/handlers/directory-vendor-handler/index.handler
+      Policies:
+        - AWSLambdaVPCAccessExecutionRole
```
Comment (shaikegross) [inline packages/vendor-directory-api/template.yml:163]: Let's move policies to global

**Review by shaikegross — APPROVED** [review]
Looks good

**Pull request merged**